gateway-js/src/index.ts

```diff
@@ -389,10 +389,9 @@ export class ApolloGateway implements GraphQLService {
       throw e;
     }
 
-    this.schema = schema;
     // TODO(trevor): #580 redundant parse
     this.parsedSupergraphSdl = parse(supergraphSdl);
-    this.queryPlanner = new QueryPlanner(schema);
+    this.updateWithSchemaAndNotify(schema);
```

Comment:
Hmm, I think the bit about `load` is a fair point but our current API encourages users to specifically _not_ handle calling `load` themselves and to let AS handle that. Still totally fine to do it the "old way", but at this point I think it's undocumented.

Is there a reason you think it's bad to call the update listeners on initial load? Noisy/not semantic maybe? I can appreciate that viewpoint.

Reply:
I mean, I agree that this API is for use by AS, and AS explicitly doesn't care about getting onSchemaChange callbacks called on the initial load (its callback does nothing if the server isn't started yet). So the fact that it's for use by AS suggests that if we want to resolve the inconsistency in the current behavior (which I think is a good idea) we should do it in the direction of fewer redundant calls rather than more?

Reply:
`onSchemaChange` is not necessarily just for AS, that could reasonably be used in userland for observability (though I see now it's undocumented). We use it for promise driven tests, so we may need to update some if its behavior is changed.

I definitely agree we should make it consistent. I also agree less redundant calls is better, though I'm not convinced this is redundant (for the userland use case).

Reply:
@trevor-scheer OK, I'm back from the long weekend, let's resolve this :) Sounds like we agree it should be consistent.

My instinct is that calling the `onSchemaChange` *before `load` resolves* with the original schema seems odd to me, and not a great fit for the "change" in its name. So I lean towards not calling it for the original load. But this is an area you are more deeply involved with, so how about you choose which way to make it consistent?

Reply:
I agree with @trevor-scheer here. While our own code only uses `onSchemaChange()` for schema reporting, it's part of the public API (albeit undocumented) and `load()` is expected to be called by Apollo Server, so I'd err on the side of exposing the initial schema load to `onSchemaChange` callbacks.

Reply:
See somewhat rambly https://github.com/apollographql/apollo-server/pull/5187#discussion_r635498292 for an idea of how to resolve this: basically, I think y'all are right that calling it always is reasonable, but let's make a more clear name for it. Coming up with a clearer name also gives us the opportunity to make this callback take an object rather than positional arguments, and to more easily detect whether the new version of the gateway is running.

(If you go through with that plan, then I think it's fine to consistently call callbacks registered either via the new name or the old name with the first schema.)